### ISSUE TITLE
Authenticating returns 404

### ISSUE THREAD

**Issue by ndaidong** [pocketbase/js-sdk#81]
Hello everybody, 

I'm investigating how to use pocketbase. Here is my `users` collection:

![Screenshot from 2022-11-22 13-38-29](https://user-images.githubusercontent.com/1809619/203242663-da70f5bb-59e7-4479-8bc6-32edb4290d66.png)

And its API rules:

![Screenshot from 2022-11-22 13-33-47](https://user-images.githubusercontent.com/1809619/203242093-7801ec5c-91bb-4a26-aa82-dce12eb13eb5.png)

At client side, I import `js-sdk` and add some code from README. 

```js
const result = await pb.collection('users').authWithPassword(email, password)
```
But it always returns 404 errror:

![Screenshot from 2022-11-22 13-41-28](https://user-images.githubusercontent.com/1809619/203243144-1ef75d2f-c778-49c2-83c5-8ac996e7ddb1.png)

What am I missing here?

Thank you for any advise.

**Comment by ganigeorgiev** [comment]
From the screenshot of the Admin UI it seems that you are using an older PocketBase version.

PocketBase JS SDK v0.8.0 is compatible only with PocketBase v0.8+ (you can find the PocketBase version you are using in the bottom right corner in the Admin UI).

For a changelog and a v0.7.x -> v0.8.0 migration guide, you could check https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#v080

**Comment by ndaidong** [comment]
@ganigeorgiev thank you, yes I'm using js-sdk@0.8.0. I've upgraded pocketbase to v0.8.0 too. Now `users` section has been replaced with the collection `profiles`. I got this point and have no comment about that. 

![Screenshot from 2022-11-22 13-54-25](https://user-images.githubusercontent.com/1809619/203246444-b59ecde8-a859-461c-a563-3d04b644dea0.png)

However latest result is still as same as my previous post, except of the error message now clearer:

![Screenshot from 2022-11-22 13-55-28](https://user-images.githubusercontent.com/1809619/203246412-77cc6cde-589b-4898-b6c9-39607c391b85.png)

I tried to change `users` to `profiles`:

```js
const result = await pb.collection('profiles').authWithPassword(email, password)
```

 It returned 400 error with message `Invalid collection type`, which is reasonable.

**Comment by ganigeorgiev** [comment]
> Now users section has been replaced with the collection profiles. I got this point and have no comment about that.

That's not correct. It should be `users`. I don't think you followed the migration guide that I linked. Please check carefully the "Migrate from v0.7.x" section - https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x
You'll need to run a special `upgrade` command against your `pb_data` before starting v0.8.0.

**Comment by ndaidong** [comment]
@ganigeorgiev ok, thank you. I've just looked at the Authentication in the new version. There is no `email/password` option as before. That maybe the cause.

**Comment by ganigeorgiev** [comment]
@ndaidong No, you just didn't run `./pocketbase upgrade`. The command will change `profiles` collection to `users`. Please read carefully the migration guide - https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x.

**Comment by ndaidong** [comment]
@ganigeorgiev I'm using docker, this is my Dockerfile. Is this correct way to migrate?

```
FROM alpine:3.16.3

ARG PB_VERSION=0.8.0

RUN apk add --no-cache unzip

# download and unzip PocketBase
ADD https://github.com/pocketbase/pocketbase/releases/download/v${PB_VERSION}/pocketbase_${PB_VERSION}_linux_amd64.zip /tmp/pb.zip
RUN unzip /tmp/pb.zip -d /pb/

EXPOSE 8090

RUN /pb/pocketbase upgrade

CMD ["/pb/pocketbase", "serve", "--http=0.0.0.0:8090"]
```

**Comment by ndaidong** [comment]
@ganigeorgiev I will try with completly new data and update the status for you.

**Comment by ganigeorgiev** [comment]
No. You need to manually run a `upgrade` command only once to upgrade your `pb_data/data.db` structure to the new one. Once again, please read the migration guide -  https://github.com/pocketbase/pocketbase/blob/master/CHANGELOG.md#migrate-from-v07x.

**Comment by ndaidong** [comment]
@ganigeorgiev Done. With new data, it works for me now. That's data conflict problem and incompleted migration as your pointed out.

Thank you so much for your help.

![Screenshot from 2022-11-22 14-16-50](https://user-images.githubusercontent.com/1809619/203249909-3c2eaac6-0c9d-4359-940d-1c5123ec0579.png)
